src/main/java/com/cloudant/client/org/lightcouch/CouchDbClient.java

```diff
@@ -1,333 +1,589 @@
+import com.cloudant.http.HttpConnectionResponseInterceptor;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.reflect.TypeToken;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.Type;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Contains a Client Public API implementation.
+ *
+ * @author Ahmed Yehia
+ * @see CouchDbClient
+ */
+
+public class CouchDbClient {
+
+    static final Log log = LogFactory.getLog(CouchDbClient.class);
+
+    private URI baseURI;
+
+    private Gson gson;
+
+    private Map<String, String> customHeaders;
+
+    private List<HttpConnectionRequestInterceptor> requestInterceptors;
+    private List<HttpConnectionResponseInterceptor> responseInterceptors;
+
+    CouchDbClient() {
+        this(new CouchDbConfig());
+    }
+
+    CouchDbClient(CouchDbConfig config) {
+        final CouchDbProperties props = config.getProperties();
+
+        this.gson = GsonHelper.initGson(new GsonBuilder()).create();
+        final String path = props.getPath() != null ? props.getPath() : "";
+        //Add username and password if authentication info exists
+        if (props.getUserInfo() != null && !props.getUserInfo().isEmpty()) {
+            this.baseURI = buildUri().scheme(props.getProtocol()).userInfo(props.getUserInfo())
+                    .host(props.getHost()).port(props.getPort())
+                    .path("/").path(path).build();
+        } else {
+            this.baseURI = buildUri().scheme(props.getProtocol()).host(props.getHost()).port(props
+                    .getPort()).path("/").path(path).build();
+        }
+
+        this.requestInterceptors = new ArrayList<HttpConnectionRequestInterceptor>();
+        this.responseInterceptors = new ArrayList<HttpConnectionResponseInterceptor>();
+
+        if(props.getRequestInterceptors() != null) {
+            this.requestInterceptors.addAll(props.getRequestInterceptors());
+        }
+
+        if(props.getResponseInterceptors() != null) {
+            this.responseInterceptors.addAll(props.getResponseInterceptors());
+        }
+
+
+        props.clearPassword();
+    }
+
+    /**
+     * Constructs a new instance of this class.
+     *
+     * @param properties An object containing configuration properties.
+     * @see {@link CouchDbProperties}
+     */
+    public CouchDbClient(CouchDbProperties properties) {
+        this(new CouchDbConfig(properties));
+    }
+
+    /**
+     * Shuts down the connection manager used by this client instance.
+     */
+    public void shutdown() {

```

Comment:
JavaDoc should be changed to reflect that we don't use a connection manager anymore